### PR TITLE
Clarify execution mode wording and add cross references.

### PR DESCRIPTION
**Pull request by castrojo**
Please take a look @liz-acosta

### CODE REVIEW

File: docs/source/aws/lambda.rst [hunk 1]
```diff
@@ -43,7 +43,7 @@ policy filters, and apply the policy actions to matching resources.
 CloudTrail API Calls
 ++++++++++++++++++++
 
-Lambdas can receive CWE over CloudTrail API calls with delay of 90s at P99.
+Lambdas can receive CWE over CloudTrail API calls with seconds of delay at P99.
```
Comment (liz-acosta) [inline docs/source/aws/lambda.rst:46]: I would call out what "CWE" stands for when it first occurs, which is line 32. That line would become

```
Custodian provides for policy level execution against any CloudWatch Events stream. Henceforth "CloudWatch Events" will be abbreviated as CWE.
```

Comment (liz-acosta) [inline docs/source/aws/lambda.rst:46]: `with seconds of delay at P99` --> `within seconds of delay at P99`

File: docs/source/aws/lambda.rst [hunk 2]
```diff
@@ -80,11 +81,15 @@ have been defined to allow for easier policy writing as for the
         event: RunInstances
         ids: "responseElements.instancesSet.items[].instanceId"
 
+Refer to the `AWS execution modes documention
+<https://cloudcustodian.io/docs/aws/resources/aws-modes.html#cloudtrail>`_ for a
+list of configuration options for this execution mode:
```
Comment (liz-acosta) [inline docs/source/aws/lambda.rst:86]: `for a list of configuration options for this execution mode:` --> `for a list of other configurable options`

File: docs/source/aws/lambda.rst [hunk 3]
```diff
@@ -103,6 +108,9 @@ Lambdas can receive EC2 instance state events in real time (seconds delay).
          - mark
          - terminate
 
+Refer to `AWS execution modes documentation
```
Comment (liz-acosta) [inline docs/source/aws/lambda.rst:111]: Same as line 84 above

**Review by liz-acosta — CHANGES_REQUESTED** [review]
Left comments for further documentation clarity

File: docs/source/aws/lambda.rst [hunk 4]
```diff
@@ -29,11 +29,14 @@ resources never became available.
...
 
-Custodian provides for policy level execution against any CWE event stream.
-Each Custodian policy can be deployed as an independent Lambda function. The
-only difference between a Custodian policy that runs in Lambda and one that
-runs directly from the CLI in poll mode is the specification of the
-subscription of the events in the mode config block of the policy.
+Custodian provides for policy level execution against any `Amazon CloudWatch
+Event
+<https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/WhatIsCloudWatchEvents.html>`_
+stream. Henceforth "CloudWatch Events" will be abbreviated as CWE. Each
```
Comment (kapilt) [inline docs/source/aws/lambda.rst:35]: s/stream/source

Comment (kapilt) [reply docs/source/aws/lambda.rst:35]: hmm actually thats the term they use upstream.. so lgtm

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**